### PR TITLE
UI 3

### PR DESCRIPTION
**Pull request by RyanKParc**

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -1,8 +1,27 @@
 import React from 'react';
 import logo from '../assets/logo.png';
```
Comment (edwinKR) [inline src/components/Header.js:2]: Let's make this assets path an absolute path rather than a relative path.

File: src/components/Header.js [hunk 2]
```diff
@@ -1,8 +1,27 @@
...
+import { auth } from '../firebase/firebase.config';
+import { removeUser } from '../redux/coderSlice';
+import { useDispatch } from 'react-redux';
+import {
+    signOut
+} from 'firebase/auth'
 
 const Header = () => {
+
+    // const userInfo = useSelector((state) => state.codershop.userInfo);
```
Comment (edwinKR) [inline src/components/Header.js:14]: Why is this code commented out? Saving it here for future reference? If not, let's go ahead and delete it.

File: src/components/Header.js [hunk 3]
```diff
@@ -1,8 +1,27 @@
 import React from 'react';
 import logo from '../assets/logo.png';
 import { Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { auth } from '../firebase/firebase.config';
+import { removeUser } from '../redux/coderSlice';
+import { useDispatch } from 'react-redux';
+import {
+    signOut
+} from 'firebase/auth'
```
Comment (edwinKR) [inline src/components/Header.js:10]: Assuming the `firebase` here is the 3rd party library where as the `firstbase` importing the `auth` is a local directory. The naming is the same so it's a bit confusing. Can we rename the `firebase` folder in your local?

File: src/components/Products.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import ProductsCard from './ProductsCard'
+
+const Products = ({ products }) => {
+    console.log(products)
```
Comment (edwinKR) [inline src/components/Products.js:5]: Remove the console logging line here, please.

File: src/components/Products.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    return (
+        <div>
+            <h1>Products</h1>
+            <div className='products-card-box'>
+                {products.map((item) => {
+                    console.log("This line", item.fields);
```
Comment (edwinKR) [inline src/components/Products.js:11]: Remove the console logging here, please.

File: src/components/Products.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+                    return <ProductsCard key={item.name} products={item.fields} />
```
Comment (edwinKR) [inline src/components/Products.js:12]: Let's make this an implicit return.

File: src/components/ProductsCard.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { useNavigate } from 'react-router-dom';
+
+const ProductsCard = ({ products }) => {
+    const navigate = useNavigate();
+    const pageId = products.name.stringValue.toLowerCase();
+    console.log(pageId)
```
Comment (edwinKR) [inline src/components/ProductsCard.js:7]: Remove the console logging.

File: src/components/ProductsCard.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { useNavigate } from 'react-router-dom';
+
+const ProductsCard = ({ products }) => {
+    const navigate = useNavigate();
+    const pageId = products.name.stringValue.toLowerCase();
```
Comment (edwinKR) [inline src/components/ProductsCard.js:6]: Are we lowercasing this b/c the react router is case sensitive when navigating to different urls?

File: src/components/Products.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import ProductsCard from './ProductsCard'
+
+const Products = ({ products }) => {
+    console.log(products)
+    return (
+        <div>
+            <h1>Products</h1>
+            <div className='products-card-box'>
+                {products.map((item) => {
```
Comment (edwinKR) [inline src/components/Products.js:10]: What are we showing if there are not products? I think that may be the edge case we have here.

File: src/components/ProductsCard.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import { useNavigate } from 'react-router-dom';
+
+const ProductsCard = ({ products }) => {
```
Comment (edwinKR) [inline src/components/ProductsCard.js:4]: Seems like `ProductsCard` is the card container for a single product. Can we remove the plural and update it to `ProductCard`. This sounds more singular and is straightforward.

File: src/index.scss [hunk 11]
```diff
@@ -80,4 +80,25 @@ header {
...
+
+.products-card-box {
+  display: grid;
+  grid-template-columns: repeat(4, minmax(0, 1fr));
+  gap: 20px;
+  margin: 0 auto;
+  max-width: 1440px;
+}
+
+.productscard{
```
Comment (edwinKR) [inline src/index.scss:93]: Fix the alignment of the css code here, please.

File: src/index.scss [hunk 12]
```diff
@@ -80,4 +80,25 @@ header {
     background-color: var(--primary-color);
     color: #fff;
   }
+}
+
+.products-card-box {
```
Comment (edwinKR) [inline src/index.scss:85]: Naming update to singular such as `product-card-box`?

File: src/pages/Home.js [hunk 13]
```diff
@@ -1,8 +1,19 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import Products from '../components/Products'
+import { useLoaderData } from 'react-router-dom'
 
 const Home = () => {
+    const [products, setProducts] = useState([]);
```
Comment (edwinKR) [inline src/pages/Home.js:6]: Formatting: const [ products, setProducts ].
Let's add a ticket to add an auto formatting on save.

File: src/pages/Home.js [hunk 14]
```diff
@@ -1,8 +1,19 @@
...
+    const data = useLoaderData();
+    console.log(data.data.documents)
```
Comment (edwinKR) [inline src/pages/Home.js:8]: Console logging.

File: src/pages/Login.js [hunk 15]
```diff
@@ -1,12 +1,17 @@
 import { useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
 import { getRedirectResult } from 'firebase/auth';
```
Comment (edwinKR) [inline src/pages/Login.js:4]: Extra space here. Let's remove it unless you did this intentionally. 
Let's add another ticket to have a fixed `imports` formatting with some standard.

File: src/pages/Login.js [hunk 16]
```diff
@@ -40,6 +45,10 @@ import SignupForm from '../components/SignupForm';
 
 const Login = () => {
 
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  // const auth = getAuth();
```
Comment (edwinKR) [inline src/pages/Login.js:50]: Commented out code.

File: src/redux/coderSlice.js [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+})
+
+export const {
+    addToCart,
+    addUser,
+    removeUser
+} = coderSlice.actions;
+export default coderSlice.reducer;
```
Comment (edwinKR) [inline src/redux/coderSlice.js:30]: Let's add a line between these two exports.

File: src/redux/store.js [hunk 18]
```diff
@@ -0,0 +1,8 @@
+import { configureStore } from '@reduxjs/toolkit'
+import coderReducer from './coderSlice'
+
+export const store = configureStore({
+    reducer: {
+        coder: coderReducer,
+    },
+})
```
Comment (edwinKR) [inline src/redux/store.js:8]: missing a semi-colon. These sort of things can be prevented if we have a nice auto-formatting on save. Once we add a ticket as mentioned earlier for this, we can research what libraries or configurations to use to make this happen.

File: src/components/ProductsCard.js (hunk 8 above)
Comment (RyanKParc) [reply src/components/ProductsCard.js:6]: yes sir

File: src/components/Products.js (hunk 9 above)
Comment (RyanKParc) [reply src/components/Products.js:10]: Is edge case by the array length a normal solution?

File: src/pages/Home.js (hunk 13 above)
Comment (RyanKParc) [reply src/pages/Home.js:6]: my auto formatting forces [products, setProducts] lol

**Pull request merged**